lib/cpp_peripheral_states.dart

```diff
@@ -0,0 +1,445 @@
+
+final logger = Logger('cpp_peripheral');
+
+class IterableExchangeState extends CppPeripheralState {
+  IterableExchangeState(
+    this.iter,
+    this.result,
+    this.cmdName,
+    this.nextState,
+  );
+
+  final Iterator<String> iter;
+  final List<String> result;
+  final String cmdName;
+  final CppPeripheralState nextState;
+
+  @override
+  void onEnter() {
+    moveIterator();
+  }
+
+  @override
+  Future<void> handlePeripheralCommand(String cmd) async {
+    if (cmd == Command.ok) {
+      logger.info('Peripheral ${cmdName} ${iter.current} supported');
+      result.add(iter.current);
+    } else if (cmd == Command.nok) {
+      logger.info('Peripheral ${cmdName} ${iter.current} unsupported');
+    } else {
+      super.handlePeripheralCommand(cmd);
+    }
+
+    moveIterator();
+  }
+
+  void moveIterator() {
+    if (iter.moveNext())
+      sendCommandToPrtipheral(join(cmdName, iter.current));
+    else
+      transitionTo(nextState);
+  }
+}
+
+class InitializedState extends CppPeripheralState {
+  @override
+  void onEnter() {
+    transitionTo(IdleState());
+    context.isCppInitialized = true;
+    sendInitializedToCentral();
+  }
+}
+
+class IdleState extends CppPeripheralState {
+  @override
+  Future<void> handlePeripheralCommand(String cmd) async {
+    if (cmd.startsWith(Command.state)) {
+      round.isMoveRejected = false;
+      round.fen = getCommandParams(cmd);
+      sendRoundUpdateToCentral();
+    } else if (cmd.startsWith(Command.err)) {
+      sendErrToCentral(getCommandParams(cmd));
+    } else if (cmd.startsWith(Command.msg)) {
+      sendMsgToCentral(getCommandParams(cmd));
+    } else if (cmd.startsWith(Command.setOption)) {
+      if (!context.cppOptions.set(getCommandParams(cmd))) {
+        sendErrToCentral('Option set failed: $cmd');
+      } else if (context.areCppOptionsInitialized) {
+        sendOptionsUpdateToCentral();
+      }
+    } else if (cmd.startsWith(Command.optionsEnd)) {
+      context.areCppOptionsInitialized = true;
+      sendOptionsUpdateToCentral();
+    } else if (cmd.startsWith(Command.optionsReset)) {
+      context.cppOptions.reset();
+      sendOptionsUpdateToCentral();
+    } else if (cmd.startsWith(Command.option)) {
+      if (!context.cppOptions.add(getCommandParams(cmd))) {
+        sendErrToCentral('Option add failed: $cmd');
+      }
+    } else {
+      await super.handlePeripheralCommand(cmd);
+    }
+  }
+
+  @override
+  Future<void> handleCentralBegin({
+    required String fen,
+    String? variant,
+    String? side,
+    String? lastMove,
+    String? check,
+    String? time,
+  }) async {
+    round.lastMove = null;
+    round.isMoveRejected = false;
+
+    if (variant != null && round.variant != variant) {
+      await sendCommandToPrtipheral(join(Command.setVariant, variant));
+      round.isVariantSupported = isVariantSupported(variant);
+      if (!round.isVariantSupported) {
+        round.variant = variant;
+        transitionTo(IdleState());
```

Comment:
we already are in idle

Reply:
But
```dart
class RoundState extends IdleState
```
`handleCentralBegin` can be called during round